lib/src/realm_class.dart

```diff
@@ -95,10 +95,13 @@ class Realm {
       throw RealmException("Object type ${object.runtimeType} not configured in the current Realm's schema."
           " Add type ${object.runtimeType} to your config before opening the Realm");
     }
-
-    final handle = metadata.class_.primaryKey == null
-        ? realmCore.createRealmObject(this, metadata.class_.key)
-        : realmCore.createRealmObjectWithPrimaryKey(this, metadata.class_.key, metadata.class_.primaryKey!);
+    RealmObjectHandle handle;
+    if (metadata.class_.primaryKey == null) {
+      handle = realmCore.createRealmObject(this, metadata.class_.key);
+    } else {
+      Object? primaryKeyValue = RealmObject.get(object, metadata.class_.primaryKey!);
```

Comment:
we should use the `accessor` to get the value. In general RealmObject.get should be called only by the generated code since this is where we have the `T` (type) information and this is compile time only. I will push the fix into the branch as we discussed.

Reply:
fixed